std/traits.d

```diff
@@ -8061,3 +8062,31 @@ enum isCopyable(S) = is(typeof(
     static assert(isCopyable!int);
     static assert(isCopyable!(int[]));
 }
+
+/**
+ * Detect whether type `V` is a vector type as defined in core.simd.
+ *
+ * Params:
+ *      V = The type to check.
+ *
+ * Returns:
+ *     `true` if `V` is a `__vector`, `false` otherwise.
+ */
+enum isSimdVector(V : __vector(T), T) = true;
+
+/// ditto
+enum isSimdVector(V) = false;
+
+///
+@nogc @safe unittest
+{
+    version(D_SIMD)
```

Comment:
Version is unnecessary.  `D_SIMD` is only for dmd-style intrinsics.

Reply:
simd types don't exist for the 32 bit platforms so i've changed to `version(X86_64)`

Reply:
> simd types don't exist for the 32 bit platforms so i've changed to version(X86_64)

Yes they do.  Or are you referring to a dmd limitation.

Reply:
without the version, on dmd -m32 gives the error.

> /phobos/std/traits.d(8087,23): Error: undefined identifier `ubyte16`

I know that the operations are not available but it also seems the alias created w/ `__vector` are not defined at all, which is surprising. Initially i though that they were defined, despite of not being usable.

Reply:
The declaration line is:
```
static if (is(Vector!(ubyte[16])))  alias Vector!(ubyte[16]) ubyte16;
```

Reply:
The right version is then `version(DigitalMars) {version(X86){} else {}}`.